src/pages/index.js

```diff
@@ -59,19 +181,60 @@ const newCardPopup = new PopupWithForm({
 });
 newCardPopup.setEventListeners();
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
 
 const profileEditPopup = new PopupWithForm({
   popupSelector: "#profile-edit-modal",
   handleFormSubmit: handleEditProfileSubmit,
 });
 
+const avatarEditModal = new PopupWithForm({
+  popupSelector: "#edit-avatar-modal",
+  handleFormSubmit: handleAvatarSubmit,
+});
+
+//Profile creation and editing
+
+const profileAvatar = document.querySelector(".profile__image-container");
+
+profileAvatar.addEventListener("click", () => avatarEditModal.open());
+
+avatarEditModal.setEventListeners();
+
+function handleAvatarSubmit({ link }) {
+  avatarEditModal.renderLoading(true);
+
+  api
+    .updateAvatar(link)
+    .then(() => {
+      userInfo.setAvatar(link);
+      avatarEditModal.close();
+    })
+    .catch((err) => {
+      console.error("Error updating avatar:", err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors